src/components/calculator.js

```diff
@@ -1,51 +1,73 @@
 import './calculator.css';
 import PropTypes from 'prop-types';
+import { useState } from 'react';
+import calculate from '../logic/calculate';
 
-const Calculator = () => (
-  <div className="calculator-app">
-    <div className="calculator">
-      <Display value="0" />
-      <div className="Digits">
-        <Button Label="AC" />
-        <Button Label="+/-" />
-        <Button Label="%" />
-        <Button Label="÷" />
-        <Button Label="7" />
-        <Button Label="8" />
-        <Button Label="9" />
-        <Button Label="x" />
-        <Button Label="4" />
-        <Button Label="5" />
-        <Button Label="6" />
-        <Button Label="-" />
-        <Button Label="1" />
-        <Button Label="2" />
-        <Button Label="5" />
-        <Button Label="+" />
-        <Button Label="0" />
-        <Button Label="." />
-        <Button Label="=" />
+const Calculator = () => {
+  const [calculatorData, setCalculatorData] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleButtonClick = (buttonName) => {
+    const updatedData = calculate(calculatorData, buttonName);
+    setCalculatorData(updatedData);
+  };
+
+  return (
+    <div className="calculator-app">
+      <div className="calculator">
+        <Display value={calculatorData.next || calculatorData.total || '0'} />
+        <div className="Digits">
+          <Button label="AC" onClick={() => handleButtonClick('AC')} />
+          <Button label="+/-" onClick={() => handleButtonClick('+/-')} />
+          <Button label="%" onClick={() => handleButtonClick('%')} />
+          <Button label="÷" onClick={() => handleButtonClick('÷')} />
+          <Button label="7" onClick={() => handleButtonClick('7')} />
+          <Button label="8" onClick={() => handleButtonClick('8')} />
+          <Button label="9" onClick={() => handleButtonClick('9')} />
+          <Button label="x" onClick={() => handleButtonClick('x')} />
+          <Button label="4" onClick={() => handleButtonClick('4')} />
```

Comment:
- [OPTIONAL] 🤔 This could have been a separate component (`Buttons`) in a separate file and rendered here. Smaller components are easy to fix, reusable, and make the app more scalable. :pray: 👍🏻